ROADMAP.md

```diff
@@ -1,10 +1,101 @@
+* Visualizations
+  * [#5668](https://github.com/kubeflow/pipelines/issues/5668) Visualize v2 metrics -- components can output metrics artifacts that are rendered in UI. [sample pipeline](https://github.com/kubeflow/pipelines/blob/307e91aaae5e9c71dde1fddaffa10ffd751a40e8/samples/test/metrics_visualization_v2.py#L103)
+  * [#3970](https://github.com/kubeflow/pipelines/issues/3970) Easier visualizations: HTML, Markdown, etc using artifact type + metadata.
+* v2 python components.
+  * A convenient component authoring method designed to support new features above natively. (v1 components do not completely support all the features mentioned above)
+  * [samples/test/lightweight_python_functions_v2_with_outputs.py](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_with_outputs.py)
+  * [samples/test/lightweight_python_functions_v2_pipeline.py](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline.py)
+* [#5669](https://github.com/kubeflow/pipelines/issues/5669) KFP semantics in MLMD
+  * MLMD state with exact KFP semantics (e.g. parameter / artifact / task names are the same as in DSL). This will enable use-cases like: “querying a result artifact from a pipeline run using MLMD API and then use the result in another system or another pipeline”.
+  
+    [Example pipeline](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline.py) and [corresponding MLMD state](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline_test.py).
+* [#5670](https://github.com/kubeflow/pipelines/issues/5670) Revamp KFP UI to show inputs & outputs in KFP semantics for v2 compatible pipelines
+* [#5667](https://github.com/kubeflow/pipelines/issues/5667) KFP data model based caching using MLMD.
+
+### KFP v2
+
+Design: [bit.ly/kfp-v2](https://bit.ly/kfp-v2)
+
+#### KFP v2 Goals
+
+* Data Management:
+  * build first class support for metadata -- recording, presentation and orchestration.
+  * making it easy to keep track of all the data produced by machine learning pipelines and how it was computed.
+* KFP native (and argo agnostic) spec and status: define a clear interface for KFP, so that other systems can understand KFP pipeline spec and status in KFP semantics.
+* Gaining more control over KFP runtime behavior: give the KFP system more control over the exact runtime behavior. This wasn’t a goal initially coming from use cases. However, the more we innovate on Data Management and KFP native spec/status, the clearer that other workflow systems become limitations to how we may implement new KFP features on our own. Therefore, re-architecturing KFP to let us get more control of runtime behavior is ideal for achieving KFP’s long term goals.
+* Be backward compatible with KFP v1: for existing features, we want to keep them as backward compatible as possible to ease upgrade.
+
```

Comment:
v2 also sets up KFP for supporting new features, such as metadata base caching, new control flow features such as fan in.

Reply:
thanks, added